instrumentation/mongo/src/test/java/brave/mongo/ITTraceMongoCommandListener.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.mongo;
+
+import brave.ScopedSpan;
+import brave.Tracing;
+import brave.propagation.ThreadLocalCurrentTraceContext;
+import brave.sampler.Sampler;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoQueryException;
+import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoClients;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.event.CommandListener;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import zipkin2.Span;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.entry;
+import static org.junit.Assume.assumeTrue;
+
+public class ITTraceMongoCommandListener {
+  private static final String A_COLLECTION_NAME = "myCollection"; // collection doesn't have to exist
+  private static final String INVALID_COLLECTION_NAME = "?.$";
+
+  private final List<Span> spans = new CopyOnWriteArrayList<>();
```

Comment:
if there is a chance for different thread start/finish (maybe? if so it should be mentioned in comments in the main code). Anyway, if there is, please use conventions we use elsewhere. I think you'll notice we tend to use blocking queue if something is async as that's more reliable. have a look at other code like kafka ITHttp etc.